### PR TITLE
Add shellcheck linter

### PR DESCRIPTION
**Pull request by albertodonato**
Run shellcheck in Travis builds.
Also, a couple of minor cleanups in the script

### CODE REVIEW

File: ubuntu-advantage [hunk 1]
```diff
@@ -1,4 +1,5 @@
 #!/bin/sh -e
+# shellcheck disable=SC2039
```
Comment (albertodonato) [inline ubuntu-advantage]: This is to disable the warning about non-POSIX use of `local` (which is fine with dash)

File: .travis.yml [hunk 2]
```diff
@@ -1,5 +1,15 @@
...
-script: tox
+script:
+  - tox
+  - shellcheck -s dash ubuntu-advantage update-motd.d/*
+addons:
+  apt:
+    sources:
+      - debian-sid
+    packages:
+      - shellcheck
```
Comment (panlinux) [inline .travis.yml]: Why do we need the package from debian?

File: ubuntu-advantage [hunk 3]
```diff
@@ -126,7 +127,7 @@ is_esm_enabled() {
 is_livepatch_enabled() {
     # it's fine if it fails because the snap isn't installed. It's still
     # a non-zero return value
-    canonical-livepatch status > /dev/null 2>&1
+    canonical-livepatch status >/dev/null 2>&1
```
Comment (panlinux) [inline ubuntu-advantage]: Is this just cosmetic?

File: ubuntu-advantage [hunk 4]
```diff
@@ -9,16 +10,16 @@ REPO_KEY_FILE="ubuntu-esm-keyring.gpg"
...
+CA_CERTIFICATES=${CA_CERTIFICATES:-"/usr/sbin/update-ca-certificates"}
+SNAPD=${SNAPD:-"/usr/lib/snapd/snapd"}
 KERNEL_VERSION=$(uname -r)
 
 check_result() {
     local result output
     result=0
     output=$("$@" 2>&1) || result=$?
-    if [ "$result" -ne "0" ]; then
+    if [ "$result" -ne 0 ]; then
```
Comment (panlinux) [inline ubuntu-advantage]: Shouldn't we also remove the quotes from $result then?

File: ubuntu-advantage (hunk 3 above)
Comment (albertodonato) [inline ubuntu-advantage]: yes

File: .travis.yml (hunk 2 above)
Comment (albertodonato) [inline .travis.yml]: shellcheck is not in precise

File: ubuntu-advantage (hunk 4 above)
Comment (albertodonato) [inline ubuntu-advantage]: yeah, I can do that too

**Comment by albertodonato** [comment]
@panlinux yeah the version in xenial seems to be quite old. I don't get them here (with 0.4.4).
We're pulling 0.4.6 from debian, FWIW

**Comment by panlinux** [comment]
ok, +1, merge away

**Pull request merged**